moonraker/components/authorization.py

```diff
@@ -933,6 +964,71 @@ def get_api_key(self) -> Optional[str]:
             return None
         return self.api_key
 
+    async def _handle_oauth2_device_flow_start(
+        self, web_request: WebRequest
+    ) -> Dict[str, Any]:
+        if self.oauth2_device_flow is None:
+            raise self.server.error("OAuth2 device flow not configured", 400)
+
+        provider_name: str = web_request.get_str('provider')
+        device_record = await self.oauth2_device_flow.initiate_device_flow(
+            provider_name
+        )
+
+        return {
+            'device_code': device_record.device_code,
+            'user_code': device_record.user_code,
+            'verification_uri': device_record.verification_uri,
+            'verification_uri_complete': (
+                device_record.verification_uri_complete
+            ),
+            'expires_in': device_record.expires_in,
+            'interval': device_record.interval
+        }
+
+    async def _handle_oauth2_device_flow_poll(
+        self, web_request: WebRequest
+    ) -> Dict[str, Any]:
+        if self.oauth2_device_flow is None:
+            raise self.server.error("OAuth2 device flow not configured", 400)
+
+        device_code: str = web_request.get_str('device_code')
+        error, provider, username = await self.oauth2_device_flow.poll_token(
+            device_code
+        )
+
+        if error:
+            return {
+                'error': error
+            }
+
+        if username not in self.users:
```

Comment:
Another one: right now we are creating a new user when someone passes the oauth2 flow, which means virtually _anyone_ would be able to access this instance with any Google account...

If using a custom realm or custom OAuth2 provider, then that would be fine I guess, as they could just control who can access that realm directly on the provider!